File: smartportfolio/history.py
"""Portfolio history module for SmartPortfolio CLI."""

from typing import List


def show_history(history: List[float]) -> None:
    """Print portfolio history as a simple chart."""
    print("\n===== Portfolio History =====")
    for i, value in enumerate(history[-10:], 1):
        bar = '*' * int(value / max(history) * 40) if history else ''
        print(f"Day {i:2d}: {bar} (${value:.2f})")
    print()

File: smartportfolio/history.js
function showHistory(history) {
  console.log('\n===== Portfolio History =====');
  const max = Math.max(...history);
  history.slice(-10).forEach((value, index) => {
    const barLength = max ? Math.floor((value / max) * 40) : 0;
    const bar = '*'.repeat(barLength);
    console.log(`Day ${String(index + 1).padStart(2, ' ')}: ${bar} ($${value.toFixed(2)})`);
  });
  console.log();
}

module.exports = { showHistory };
